ttls/client.py

```diff
@@ -76,25 +76,72 @@ async def _post(self, endpoint: str, **kwargs) -> Any:
         if "json" in kwargs:
             logger.info("POST payload %s", kwargs["json"])
         headers = kwargs.pop("headers", self.headers)
-        async with self.session.post(
-            f"{self.base}/{endpoint}", headers=headers, **kwargs
-        ) as r:
-            return await r.json()
+        retry_num = kwargs.pop("retry_num", 0)
+        try:
+            async with self.session.post(
+                f"{self.base}/{endpoint}", headers=headers, **kwargs
+            ) as r:
+                return await r.json()
+        except ClientResponseError as e:
+            if e.status == 401:
+                max_retries = 1
+                if retry_num < max_retries:
+                    retry_num += 1
+                    logger.debug(
+                        f"Invalid token for POST. Refreshing token and attempting retry {retry_num} of {max_retries}."
+                    )
+                    await self.refresh_token()
+                    await self._post(
+                        endpoint, headers=self.headers, retry_num=retry_num, **kwargs
+                    )
+                else:
+                    logger.debug(
+                        f"Invalid token for POST. Maximum retries of {max_retries} exceeded."
+                    )
+                    raise e
+            else:
+                raise e
 
     async def _get(self, endpoint: str, **kwargs) -> Any:
         await self.ensure_token()
         logger.info("GET endpoint %s", endpoint)
         headers = kwargs.pop("headers", self.headers)
-        async with self.session.get(
-            f"{self.base}/{endpoint}", headers=headers, **kwargs
-        ) as r:
-            return await r.json()
+        retry_num = kwargs.pop("retry_num", 0)
+        try:
+            async with self.session.get(
+                f"{self.base}/{endpoint}", headers=headers, **kwargs
+            ) as r:
+                return await r.json()
+        except ClientResponseError as e:
```

Comment:
This is the same code as above, can you refactor so this it is handled with common code?